### DISCUSSION TITLE
JavaScript libraries cannot work when using default browser

### DISCUSSION THREAD

**Discussion by davidleetw** [eclipse-rap/org.eclipse.rap#254] [Q&A]
Dear all,

By using Remote Object/BrowserFunction, I made some plugins
integrating with JavaScript libraries like Signature Pad, QR Code Scanner.

No matter at design time or at run time, all work well while executing on external browsers like Chrome/Edge/Firefox, but never work on Eclipse IDE's default internal browser.

A simple test:
Attached zipped file is from Signature Pad official web site, cannot sign if open index.html inside Eclipse IDE and default internal browser, but works well if open with external browsers mentioned above.

How to work around this issue?
To let Eclipse IDE's default internal browser behave as same as external browsers for that.

Thank you very much.

Environment:
Eclipse 4.17/4.20
RAP 3.17
Java 11

Best regards,
David

**Comment by ifurnadjiev** [comment]
Internal Eclipse IDE (SWT) browser is limited it terms of security and support. Why do you need the internal browser support? You can make the external Chrome or Firefox default Eclipse IDE browser and use it during the development. Are there any other requirements to use the SWT browser?

**Reply by davidleetw** [reply]
Hi, Ivan

Nice to see you again here! 

I built an QR Code scanner plugin by using RemoteObject, but I also need to use this plugin on an SWT desktop application, that's why I need default browser when using SWT Browser widget! A little bit something like single-sourcing concept!

This subject might not a RAP one, but I don't know how to proceed it!
Thank you and have a nice day!

Best Regards
David
